libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -61,17 +74,18 @@ namespace pika::mpi::experimental {
         /// This should be used with great caution as setting it too low can
         /// cause deadlocks. The default value is size_t(-1) - i.e. unlimited
         /// The value can be set using an environment variable as follows
-        /// PIKA_MPI_MSG_THROTTLE=512
+        /// PIKA_MPI_MSG_THROTTLE=64
         /// but user code setting it will override any default or env value
         /// This function returns the previous throttling threshold value
-        PIKA_EXPORT size_t set_max_requests_in_flight(size_t);
+        PIKA_EXPORT std::uint32_t set_max_requests_in_flight(
+            std::uint32_t, stream_index = stream_index(-1));
 
         /// Query the current value of the throttling threshold
-        PIKA_EXPORT size_t get_max_requests_in_flight();
+        PIKA_EXPORT std::uint32_t get_max_requests_in_flight(stream_index);
 
         // -----------------------------------------------------------------
         /// returns the number of mpi requests currently outstanding
-        PIKA_EXPORT size_t get_num_requests_in_flight();
+        PIKA_EXPORT std::uint32_t get_num_requests_in_flight();
```

Comment:
Why `uint32_t` instead of `size_t`? Not that we'll probably ever have more than `2^32` requests in flight at the same time... but still.

Reply:
because in your changes, you changed one of my `size_t` into a `uint32_t` so I thought for some reason you preferred it. Perhaps that was a victim of my conflict resolution. I originally had `size_t` for them all. (though frankly 64 bits is probably overkill for these vars)

Reply:
If that's the case, my bad! I don't have any memory of changing them to `uint32_t`s... Do you have a link to what I changed? In any case, `uint32_t` or `size_t` are both fine by me.